lib/core/request.js

```diff
@@ -54,19 +54,22 @@ try {
 }
 
 class Request {
-  constructor (origin, {
-    path,
-    method,
-    body,
-    headers,
-    query,
-    idempotent,
-    blocking,
-    upgrade,
-    headersTimeout,
-    bodyTimeout,
-    throwOnError
-  }, handler) {
+  constructor (origin, options, handler) {
+    const {
+      path,
+      method,
+      body,
+      headers,
+      query,
+      idempotent,
+      blocking,
+      upgrade,
+      headersTimeout,
+      bodyTimeout,
+      reset,
+      throwOnError
+    } = options
```

Comment:
Why change this?

Reply:
Do you mean the reset?

Mostly to validate the `reset` value and carry over the setting during the request lifecycle, as it is attached to a single request.

Another option is to pass it as a third-parameter to the `write` function on `lib/client`. What are your thoughts? 🤔

Reply:
Just add `reset` to the previous list. I don't understand why you moved out options.